--- remote 
+++ local 
@@ -17,6 +17,7 @@
 //-----------------------------------------------------------------------
 namespace OdinSerializer
 {
+#if UNITY_RUNTIME
     using System;
     using System.Collections;
     using System.Collections.Generic;
@@ -63,7 +64,7 @@
         /// The dictionary keys to remove.
         /// </summary>
         public object[] DictionaryKeysRemoved;
-
+        
         /// <summary>
         /// Applies the modification to the given Object.
         /// </summary>
@@ -300,6 +301,7 @@
             }
         }
 
+
         private static void ReplaceAllReferencesInGraph(object graph, object oldReference, object newReference, HashSet<object> processedReferences = null)
         {
             if (processedReferences == null)
@@ -653,4 +655,5 @@
             }
         }
     }
+#endif
 }
\ No newline at end of file
