--- remote
+++ local
@@ -92,8 +92,11 @@
 
 #if UNITY_EDITOR
         private static Assembly EditorAssembly = typeof(UnityEditor.Editor).Assembly;
-#endif
+#elif UNITY_RUNTIME
         private static Assembly EngineAssembly = typeof(UnityEngine.Object).Assembly;
+#else
+        private static Assembly EngineAssembly = null;
+#endif
 
         private static bool EmitIsIllegalForMember(MemberInfo member)
         {
