--- remote 
+++ local 
@@ -20,7 +20,9 @@
     using System;
     using System.Collections;
     using System.Collections.Generic;
+#if UNITY_RUNTIME
     using UnityEngine;
+#endif
 
     /// <summary>
     /// Interface for immutable list.
@@ -46,7 +48,9 @@
     [Serializable]
     public sealed class ImmutableList : IImmutableList<object>
     {
+#if UNITY_RUNTIME
         [SerializeField]
+#endif
         private IList innerList;
 
         /// <summary>
@@ -255,7 +259,9 @@
     [Serializable]
     public sealed class ImmutableList<T> : IImmutableList<T>
     {
+#if UNITY_RUNTIME
         [SerializeField]
+#endif
         private IList<T> innerList;
 
         /// <summary>
