--- /home/joeclack/RiderProjects/OdinSerializerThunderstore/submodules/odin-serializer/OdinSerializer/OdinSerializer.csproj	2024-02-26 10:57:39.539857106 +0000
+++ /home/joeclack/RiderProjects/OdinSerializerThunderstore/OdinSerializer/OdinSerializer.csproj	2024-02-26 13:00:42.578468044 +0000
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project Sdk="Microsoft.NET.Sdk">
   <Import Project="..\Libraries\MSBuild\15.0\Microsoft.Common.props" Condition="Exists('..\Libraries\MSBuild\15.0\Microsoft.Common.props')" />
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
@@ -9,9 +9,8 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>OdinSerializer</RootNamespace>
     <AssemblyName>OdinSerializer</AssemblyName>
-    <TargetFrameworkVersion>v3.5</TargetFrameworkVersion>
+    <TargetFramework>netstandard2.0</TargetFramework>
     <FileAlignment>512</FileAlignment>
-    <TargetFrameworkProfile>Unity Subset v3.5</TargetFrameworkProfile>
     <ProjectTypeGuids>{E097FAD1-6243-4DAD-9C02-E9B9EFC3FFC1};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
   </PropertyGroup>
   <PropertyGroup>
@@ -19,7 +18,7 @@
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug Editor|AnyCPU'">
     <OutputPath>..\Build\Assets\Plugins\Sirenix\Odin Serializer\EditorOnly\</OutputPath>
-    <DefineConstants>DEBUG;CAN_EMIT;UNITY_EDITOR;ENABLE_PROFILER</DefineConstants>
+    <DefineConstants>DEBUG;CAN_EMIT;UNITY_EDITOR;ENABLE_PROFILER;UNITY_RUNTIME</DefineConstants>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <NoWarn>CS1723, CS1723, CS1574, CS1584, CS1658</NoWarn>
     <DocumentationFile>..\Build\Assets\Plugins\Sirenix\Odin Serializer\EditorOnly\OdinSerializer.xml</DocumentationFile>
@@ -32,7 +31,7 @@
     <OutputPath>..\Build\Assets\Plugins\Sirenix\Odin Serializer\EditorOnly\</OutputPath>
     <PlatformTarget>AnyCPU</PlatformTarget>
     <NoWarn>CS1723, CS1723, CS1574, CS1584, CS1658</NoWarn>
-    <DefineConstants>DEBUG;CAN_EMIT;UNITY_EDITOR;ENABLE_PROFILER</DefineConstants>
+    <DefineConstants>DEBUG;CAN_EMIT;UNITY_EDITOR;ENABLE_PROFILER;UNITY_RUNTIME</DefineConstants>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <Optimize>true</Optimize>
     <DocumentationFile>..\Build\Assets\Plugins\Sirenix\Odin Serializer\EditorOnly\OdinSerializer.xml</DocumentationFile>
@@ -44,6 +43,19 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release JIT|AnyCPU'">
     <OutputPath>..\Build\Assets\Plugins\Sirenix\Odin Serializer\JIT\</OutputPath>
     <NoWarn>CS1723, CS1723, CS1574, CS1584, CS1658</NoWarn>
+    <DefineConstants>CAN_EMIT;UNITY_RUNTIME</DefineConstants>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <DocumentationFile>
+    </DocumentationFile>
+    <Optimize>true</Optimize>
+    <LangVersion>4</LangVersion>
+    <ErrorReport>prompt</ErrorReport>
+    <DebugType>pdbonly</DebugType>
+    <DebugSymbols>true</DebugSymbols>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release No-Unity JIT|AnyCPU'">
+    <OutputPath>..\Build\Assets\Plugins\Sirenix\Odin Serializer\No-Unity JIT\</OutputPath>
+    <NoWarn>CS1723, CS1723, CS1574, CS1584, CS1658</NoWarn>
     <DefineConstants>CAN_EMIT</DefineConstants>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <DocumentationFile>
@@ -57,8 +69,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release AOT|AnyCPU'">
     <OutputPath>..\Build\Assets\Plugins\Sirenix\Odin Serializer\AOT\</OutputPath>
     <NoWarn>CS1723, CS1723, CS1574, CS1584, CS1658</NoWarn>
-    <DefineConstants>
-    </DefineConstants>
+    <DefineConstants>UNITY_RUNTIME</DefineConstants>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <DocumentationFile>
     </DocumentationFile>
@@ -87,16 +98,19 @@
       <Private>False</Private>
     </Reference>
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework.TrimEnd(`0123456789`))' == 'net'">
+    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" PrivateAssets="all"/>
+  </ItemGroup>
   <ItemGroup>
     <Compile Include="*.cs;.\**\*.cs" />
     <Compile Remove="obj\**" />
   </ItemGroup>
-  <ItemGroup />
+  <ItemGroup Condition="$(Configuration) == 'Release No-Unity JIT'">
+    <Compile Remove="Unity Integration\**" />
+    <PackageReference Include="JetBrains.Annotations" Version="2023.3.0" PrivateAssets="all" />
+    <Reference Remove="Unity*" />
+  </ItemGroup>
   <Import Project="..\Libraries\MSBuild\15.0\Bin\Microsoft.CSharp.targets" />
-  <PropertyGroup>
-    <PostBuildEvent>call "$(SolutionDir)Libraries\PDB2MDB\pdb2mdb.exe" "$(TargetPath)"
-call del /a "$(TargetDir)*.pdb*"</PostBuildEvent>
-  </PropertyGroup>
   <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
        Other similar extension points exist, see Microsoft.Common.targets.
   <Target Name="BeforeBuild">
